python/ray/dashboard/client/src/pages/dashboard/tune/TuneTable.tsx

```diff
@@ -0,0 +1,219 @@
+import { Theme } from "@material-ui/core/styles/createMuiTheme";
+import createStyles from "@material-ui/core/styles/createStyles";
+import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
+import React from "react";
+import { connect } from "react-redux";
+import { StoreState } from "../../../store";
+import { dashboardActions } from "../state";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      padding: theme.spacing(2),
+      "& > :not(:first-child)": {
+        marginTop: theme.spacing(2)
+      }
+    },
+    table: {
+      marginTop: theme.spacing(1)
+    },
+    cell: {
+      padding: theme.spacing(1),
+      textAlign: "center",
+      "&:last-child": {
+        paddingRight: theme.spacing(1)
+      }
+    },
+    warning: {
+      fontSize: "0.8125rem",
+      marginBottom: theme.spacing(2)
+    },
+    warningIcon: {
+      fontSize: "1.25em",
+      verticalAlign: "text-bottom"
+    }
+  });
+
+const mapStateToProps = (state: StoreState) => ({
+  tuneInfo: state.dashboard.tuneInfo
+});
+
+const mapDispatchToProps = dashboardActions;
+
+class TuneTable extends React.Component<
+  WithStyles<typeof styles> &
+    ReturnType<typeof mapStateToProps> &
+    typeof mapDispatchToProps
+> {
+  timeout: number = 0;
+
+  state = {
+    sortedColumn: "",
+    ascending: true,
+    metric: true,
+  };
+
+  onColumnClick = (column: string, metric: boolean) => {
+    let ascending = this.state.ascending;
+    if (column === this.state.sortedColumn) {
+      ascending = !ascending;
+    } else {
+      ascending = true;
+    }
+    this.setState({
+      sortedColumn: column,
+      ascending: ascending,
+      metric: metric
+    });
+  };
+
+  sortedCell = (name: string, metric: boolean) => {
+    const { tuneInfo, classes } = this.props
+    const { sortedColumn, ascending } = this.state;
+    let label: "desc" | "asc" = "asc";
+
+    if (name === sortedColumn && !ascending) {
+      label = "desc";
+    }
+
+    if (tuneInfo === null) {
+      return;
+    }
+
+    const type = (metric ? "metrics" : "params");
+    const key  = Object.keys(tuneInfo["trial_records"])[0]
+    const first_item = tuneInfo["trial_records"][key][type][name]
+    if (isNaN(parseFloat(first_item))) {
+      return  (
+        <TableCell className={classes.cell}> {name} </TableCell>
+      )
+    }
+
+    return (
+      <TableCell
+        className={classes.cell}
+        key={name}
+        onClick={() => this.onColumnClick(name, metric)}
+      >
+        <TableSortLabel active={name === sortedColumn} direction={label}>
```

Comment:
Can we fix the centering issue either by left-aligning text and moving the arrow to the right, or right-aligning numbers and moving the arrow to the left? This might be taken care of automatically when we specify the `align` property on the `TableCell`. See https://material-ui.com/components/tables/#sorting-amp-selecting for some example code (click the `< >` button below the example to show the source code).